### PR TITLE
Lib merge

### PR DESCRIPTION
**Pull request by dstathis**
Update for using the new version of loki_push_api.py.

### CODE REVIEW

File: metadata.yaml [hunk 1]
```diff
@@ -26,11 +26,11 @@ requires:
     interface: prometheus_remote_write
   metrics-endpoint:
     interface: prometheus_scrape
-  logging:
+  logging-consumer:
     interface: loki_push_api
 
 provides:
-  log_proxy:
+  logging-provider:
```
Comment (simskij) [inline metadata.yaml:33]: Can this be made less ambiguous using other names? I'm thinking "write-logs" and "read-logs" or similar. It adds no additional value to use the exact term used as the parent key name.

File: src/charm.py [hunk 2]
```diff
@@ -58,8 +58,10 @@ def __init__(self, *args):
         self._remote_write = PrometheusRemoteWriteConsumer(self)
         self._scrape = MetricsEndpointConsumer(self)
 
-        self._loki_consumer = LokiPushApiConsumer(self)
-        self._log_proxy = LogProxyProvider(self)
+        self._loki_consumer = LokiPushApiConsumer(self, relation_name="logging-consumer")
```
Comment (simskij) [inline src/charm.py:61]: Rename to `LokiPushApiRequirer` throughout

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 3]
```diff
@@ -277,21 +392,48 @@ def __init__(self, *args):
...
+PROMTAIL_BINARY_ZIP_URL = (
+    "https://github.com/grafana/loki/releases/download/v2.4.1/promtail-linux-amd64.zip"
+)
+
+
+# Paths in `charm` container
+BINARY_DIR = "/tmp"
+BINARY_ZIP_FILE_NAME = "promtail-linux-amd64.zip"
+BINARY_ZIP_PATH = "{}/{}".format(BINARY_DIR, BINARY_ZIP_FILE_NAME)
+BINARY_FILE_NAME = "promtail-linux-amd64"
```
Comment (simskij) [inline lib/charms/loki_k8s/v0/loki_push_api.py:412]: Needs to also support arm64.

File: lib/charms/loki_k8s/v0/loki_push_api.py [hunk 4]
```diff
@@ -842,11 +1037,70 @@ def alerts(self) -> dict:
...
+        if invalid_files:
+            must_contain = ["'alert'", "'expr'"]
+            if not self.allow_free_standing_rules:
+                must_contain.append("'%%juju_topology%%'")
+
+            message = "Failed to read alert rules (must contain {}): ".format(
+                ", ".join(must_contain)
+            ) + ", ".join(map(str, invalid_files))
+        elif not alert_groups:
+            message = "No alert rules found in {}".format(self._alert_rules_path)
```
Comment (simskij) [inline lib/charms/loki_k8s/v0/loki_push_api.py:1095]: `%%juju_topology%%` is not a requirement. However, the consumer needs to strap on labels containing the juju topology, and the provider needs to read them and use `promql_transform` to inject them into the expression.

**Review by simskij — CHANGES_REQUESTED** [review]

**Comment by simskij** [comment]
Feel free to create separate tickets for the issues I mentioned in the review and let me know, and I'll approve it, given that the things I point out mainly fall outside of the scope of the actual merge.

File: lib/charms/loki_k8s/v0/loki_push_api.py (hunk 4 above)
Comment (sed-i) [reply lib/charms/loki_k8s/v0/loki_push_api.py:1095]: I refactored all that in https://github.com/canonical/loki-k8s-operator/pull/79

**Review by sed-i — APPROVED** [review]

File: src/charm.py (hunk 2 above)
Comment (dstathis) [reply src/charm.py:61]: As discussed today we are holding off on this for consistency.

**Review by rbarry82 — APPROVED** [review]

**Review by simskij — APPROVED** [review]

**Pull request merged**